masq_lib/src/percentage.rs

```diff
@@ -80,178 +229,418 @@ impl Percentage {
+            (51, RoundingTo::BiggerPositive, RoundingTo::BiggerNegative),
+            (5, RoundingTo::SmallerPositive, RoundingTo::SmallerNegative),
+            (
+                100,
+                RoundingTo::SmallerPositive,
+                RoundingTo::SmallerNegative,
+            ),
+            (
+                787879,
+                RoundingTo::BiggerPositive,
+                RoundingTo::BiggerNegative,
+            ),
+            (
+                898784545,
+                RoundingTo::SmallerPositive,
+                RoundingTo::SmallerNegative,
+            ),
         ]
         .into_iter()
-        .for_each(|(num, expected_result)| {
-            assert_eq!(Percentage::should_be_rounded_down(num), expected_result)
-        })
+        .for_each(
+            |(num, expected_result_for_unsigned_base, expected_result_for_signed_base)| {
+                let result = PurePercentage::should_be_rounded_to(num, 100);
+                assert_eq!(
+                result,
+                expected_result_for_unsigned_base,
+                "Unsigned number {} was identified for rounding as {:?} but it should've been {:?}",
+                num,
+                result,
+                expected_result_for_unsigned_base
+            );
+                let signed = num as i64 * -1;
+                let result = PurePercentage::should_be_rounded_to(signed, 100);
+                assert_eq!(
+                result,
+                expected_result_for_signed_base,
+                "Signed number {} was identified for rounding as {:?} but it should've been {:?}",
+                signed,
+                result,
+                expected_result_for_signed_base
+            )
+            },
+        )
     }
 
     #[test]
     fn add_percent_to_works() {
-        let percentage = Percentage::new(13);
+        let subject = PurePercentage::try_from(13).unwrap();
 
-        let result = percentage.add_percent_to(100);
+        let unsigned = subject.add_percent_to(100);
+        let signed = subject.add_percent_to(-100);
 
-        assert_eq!(result, 113)
+        assert_eq!(unsigned, 113);
+        assert_eq!(signed, -113)
     }
 
     #[test]
-    #[should_panic(expected = "Overflowed during addition of 1 per cent, that is \
+    #[should_panic(expected = "Overflowed during addition of 1 percent, that is \
     184467440737095516, to 18446744073709551615 of type u64.")]
     fn add_percent_to_hits_overflow() {
-        let _ = Percentage::new(1).add_percent_to(u64::MAX);
+        let _ = PurePercentage::try_from(1)
+            .unwrap()
+            .add_percent_to(u64::MAX);
     }
 
     #[test]
     fn subtract_percent_from_works() {
-        let percentage = Percentage::new(55);
+        let subject = PurePercentage::try_from(55).unwrap();
 
-        let result = percentage.subtract_percent_from(100);
+        let unsigned = subject.subtract_percent_from(100);
+        let signed = subject.subtract_percent_from(-100);
```

Comment:
When I looked the assertion of the signed number, I was wrapped under the impression that the number will end up as `-155`. So, I got confused with the word subtract.

So, a better name for the function `subtract_percent_from()` will be `decrease_by_percent_for()`. Similarly, you'd like to rename the sibling function `add_percent_from()` to `increase_by_percent_for()`.